### DISCUSSION TITLE
IOS Companion App to App Store?

### DISCUSSION THREAD

**Discussion by benjycov** [mKeRix/room-assistant#688] [Q&A]
Is the companion app on a journey to be available on the App Store?

While TestFlight is okay for me, Apple restrict childrens accounts to age 13+, hence I can't deploy the companion app to their iPads to track where they are in the house!

**Comment by mKeRix** [comment]
Interesting, I didn't know about that limitation. The plan definitely is to get the app onto the real App Store. I haven't submitted it so far since I assumed that only the Bluetooth part would not be enough content to meet the App Store guidelines. In the last days I have put some more time into the app and built a little instance/entity browser for it, which I hope would be enough to convince the review team that the app brings value. I think I'll do some more minor adjustments and then try to submit it, let's see what comes out of it!

**Reply by mKeRix** [reply]
@benjycov Good news! The app went through the review process without complications and has just been released on the App Store. Note that it may take up to 24h to appear in all regions. You should be able to download it here: https://apps.apple.com/de/app/room-assistant/id1538642237

I will update the docs with the released link.

**Reply by neofraz03** [reply]
> @benjycov Good news! The app went through the review process without complications and has just been released on the App Store. Note that it may take up to 24h to appear in all regions. You should be able to download it here: https://apps.apple.com/de/app/room-assistant/id1538642237
> 
> I will update the docs with the released link.

Looks like it is no longer available on the AppStore... trying to track wife's iPhone.  Any update on this?

**Comment by benjycov** [comment]
That’s most excellent news indeed! Yes it’s not appearing here in the UK
yet but I will keep an eye out for it.

Thanks for sorting - it will be useful for finding the kids when they hide!

The new browser section - what is it supposed to do?

Regards… Ben

**Comment by benjycov** [comment]
Okay - it is on the App Store if I use the link.

Also I figured out the browser section, (I forgot to add the port). That’s
actually really useful for tuning.

It is giving me a connection error though when I click through to the
instance:


Not sure why?!

Regards… Ben

**Reply by mKeRix** [reply]
The instance browser is meant to be used with room-assistant v2.17.0 (just recently released) or higher, so any version before that might cause some issues. With v2.17.0 you should also see all instances on your network appear in the app automatically (given that you haven't turned autoDiscovery off).

**Comment by benjycov** [comment]
Ahh that explains it, I will roll the update out now.

**Comment by benjycov** [comment]
Wow this is awesome, thank you so much!

Not only can I now find the kids/their iPads, but also being able to see
the distances reported live on a device makes tuning soo much easier.

**Comment by stats2909** [comment]
Amazing stuff, this project is so cool, thank you so much.

I have this working 'mostly' but getting some odd results.  while i try to debug whats happening, can I ask a few questions?
1) should the companion app browser discover both Pi Zeros and HA add-ins (given that all on version 2.17.0 and above)
2) do i need to activate the http api port in the HA add-in config menu
3) why IP addresses over host names /FQDNs (internally resolved)
4) if i explicitly enter the IPs/FQDN of peers do i do it in all instance config files?
5) should i have a cluster for each instance that contains all other instances (ie 6 clusters each with 6 instances)?
6) woudl haveing more than one HA instance with RA add on install mess things up?  (i have 3 - main has no mqtt specified, the other 2 explicitly specifiy the first as the mqtt broker)

Thanks a million :)

**Reply by benjycov** [reply]
Hi,

I will attempt to answer your questions:

1) I have manually added my main HA instance to the browser on the companion app, it then auto discovers the last PI0 on my network. Ultimately you only need one node showing here as they all show the same data and its just to help with debugging/calibrating your setup.

2) yes. 6415 for HTTP API, 6425 for Internal communication between devices.

3) both should work if internal DNS on your network works.

4) Yes all instances need to have all neighbours listed in the cluster config.

5) No you should run one cluster containing all your nodes/instances.

6) It should work just fine, ultimately the RA install is piggy backing on the HA install, they then all talk behind the scenes to whichever HA instance you have running your MQTT broker. i.e. the RA add-on doesn't automatically talk directly to the HA instance it is installed on.

I hope that helps.

Ben

**Reply by stats2909** [reply]
Awesome, thanks so much!

Clearly Im doing something wrong... its al works (mostly).. but i get a many clusters as i do instances...

I will have a play to see if i can improve things!

**Reply by dtdmdrums** [reply]
Check out the Cluster documentation. You probably need to set a quorum. https://www.room-assistant.io/guide/cluster.html#usage-tips

**Reply by stats2909** [reply]
Have done that as part of my experimenting.  System seems to be better.  Biggest problem is the inconsistency.  Sometimes it's perfect, sometime the Pi Zero W's just seem to go to sleep.

Biggest issue though seems to be that while the companion app reports things correctly Home assistant isnt getting the Room updates via MQTT (looking at the MQTT trails the distance and time stamps are updated, but not the room).

This means the companion app is right but HA is wrong.  No idea how to change that as its deep in the guts of the system.

**Reply by mKeRix** [reply]
@stats2909 Feel free to open a new issue with some more details on that, maybe we can figure out the problem. (if you already opened one: maybe give it a bump - sorry, my inbox is a mess right now)

**Reply by stats2909** [reply]
Unable to deploy ansible playbooks due to errors with that (issue raised in you ansible forum).  Any ideas?

**Comment by JeroenTuinstra** [comment]
In the bug section I mentioned that there are problems with the companion app and iOS 15.0.1 [https://github.com/mKeRix/room-assistant/issues/931](url). 

Somehow on my Iphone 11, I have to keep room assistant app open for it to keep updating. If i close the app after several minutes the nodes do not receive anymore updates and it is reported as "not_home" by the node, which is dutifully reported by Home Assistant. 

Is anyone else experiencing problems with the app? Also when clicking on entities very often nothing shows up, or only one after awhile and then the rest. But very sketchy. 

If it worked and was reliable it would be really great, right now I can hardly base any automation on the app, as it reports it often "not_home" while i am close to one of the 4 nodes in the house.

**Comment by JeroenTuinstra** [comment]
@mKeRix I guess the app and the software is no longer active? For several weeks now there is no response to any issue or bug. I have given up on the app for BT low energy, it is completely unreliable - and it seems nobody cares. 
It looked like something really promising but the app is just very inaccurate.

**Reply by stats2909** [reply]
Personally I think thats a bit harsh! Let’s not forget this is free and no commitments are given by the dev who is doing this on the side of his actual job.

The last release has definitely been more reliable. The only issue I have with BLE is that HA reports as unavailable seemingly on a repeated and very regular (every 2-3 mins) for about 30s.

Ive combed through the setting, reboot my pi Zeros every night but I’m sure I’m missing something, the companion app shows the status flawlessly except for when it’s squiring then new MAC address and occasionally reports not_home (easy to deal with in HA)  Is it perfect, no. Am I paying for perfect, no. Will moving to wired vs wireless connectivity help. It seems to on my test batch. Is it way better than anything I could do myself, definitely yes.  Do I think I can work with what I’ve got to create a solution, again yes and I’ll learn in the way, win win!

I for one, thank @mKeRix for all his efforts, and the others that chime in to help. Keep up the good work

**Reply by JeroenTuinstra** [reply]
Okay, I agree, maybe I am a little bit harsh. But I haven't been able to get it working reliable at all. At least the iOS app seems to only update when you open it, which is no use. Then it would report wrong distances and location the whole time. See issue here: [https://github.com/mKeRix/room-assistant/issues/931](url), and no response whatsoever.
And being unavailable for 30 seconds every 2 - 3 minutes, is unreliable and turns it into less useful than BT classic, which doesn't have that problem.
You are right we cannot demand any more quality than we pay for. I appreciate the work @mKeRix is doing, but on the website room-assistant.io you get the impression of a good working piece of software .... which it isn't.

**Reply by stats2909** [reply]
So,  not sure if this is going to help you or not, but was suffering similar issues to those that it sounds like you are having.

And now I've a system which seems pretty good.  In fact, last time it went down it was because the bluetooth on my iphone gave up the ghost... a huge change in stability I think you'll agreed!

How did I achieve this? 

1) really review those config files, there can be gremlins in them, no matter how hard you look.
2) POE / wire connect as many pi Zeros as possible
3) only use one B.Tooth integration
4) download somethign like MQTT explorer to really see whats happening
5) use the quorum feature and set it to (# of instances /2) +1 

Finally what made mine work?  It seems that two of the instances were just unreliable.  I located them by looking at the response times and MQTT traces and then, by the letter rebuilt them according to the guides in this wiki.

And now (touch wood) its better than its ever been and, beyond the usual 'tech foibles' more than good enough to consider building automations on the back off.

@mKeRix and anyone else contributing to this.... awesome work guys, really REALLY appreciate it! you guys rock :)

**Reply by JeroenTuinstra** [reply]
@stats2909 Thanks for the suggestions. I have switched to bluetooth Classic. The BLE actually worked, however the application has to be open all the time (open or in the app switcher, i.e. "backgroud"). It is not an app that run truly in the background, which was for me a deal breaker. It is just to impractical to have that app open all the time.
But classic is working great. It is just a little bit slow that is all.
